### DISCUSSION TITLE
How to configure multi-language sites?

### DISCUSSION THREAD

**Discussion by JetGum** [datenstrom/community#558] [Ask a question] [answered]
Dears,

hope you're all safe and healthy!

I startetd developing a site with two languages DE/EN and find that existing documentation on how to implement a multilanguage site is a bit over my head... apart from setting CoreMultiLanguageMode to 1. 

Contentwise, everything is clear - under content/, I have content/1-en and content/2-de. If I manually change URLs, the content is displayed in the respective language.

What I cannot figure out is this: 

1. How to implement a language switch situated in navigation bar (navigation-tree.html). What I am looking for is a little anchor text "EN | DE" linking to the respective URL / language. 

https://datenstrom.se/ uses the (sub-optimal) link to a page where two links Deutsch | English can be found. I have a hunch that these are links to collection of pages? This is my navigation snippet:

```<?php list($name, $pages, $level) = $this->yellow->getLayoutArguments() ?>
<?php if (!$pages) $pages = $this->yellow->content->top() ?>
<?php $this->yellow->page->setLastModified($pages->getModified()) ?>
<?php if (!$level): ?>
<div class="nav">
   <button aria-expanded="false" aria-controls="nav__burgermenu" class="button">
        <span class="open">&#9776;</span>
        <span class="close">x</span>
        Navigation</button>
    <?php endif ?>
    <ul class="nav__menu">
        <?php foreach ($pages as $page): ?>
        <?php $children = $page->getChildren() ?>
        <li class="nav__item">
            <a<?php echo $page->isActive() ? " class=\"nav__item--active\" aria-current=\"page\"" : "" ?>
                href="<?php echo $page->getLocation(true) ?>">
                <?php echo $page->getHtml("titleNavigation") ?></a><?php if ($children->count()) { echo "\n" ; $this->yellow->layout($name, $children, $level+1);  } ?>
        </li>
        <?php endforeach ?>
    </ul>
    <?php if (!$level): ?>
</div>
```

2. What happens to shared files? In a monolingual environment, I had a footer.md in /content/shared/, but this does not work any more. The content-div is created, since it is part of footer.html, but no content is shown (suspecting that the code looks in the wrong place, which is not content/shared/).

```
<footer>

    <?php if ($page = null) $this->yellow->page->setPage("footer", $page) ?>
    <?php if ($this->yellow->page->isPage("footer")) echo $this->yellow->page->getPage("footer")->getContent() ?>

</footer>
<?php echo $this->yellow->page->getExtra("footer") ?>

```

Any hints appreciated!

Cheers,

Carsten

**Comment by JetGum** [comment, marked as answer]
Issue 1 can be solved by an individual layout files for each language. header.html can then include a language switcher.
Issue 2 can be solved by having a /shared/ folder for language. Folder needs to be within /content/language, e.g. /content/1-en/shared/.

Can be closed!

**Comment by JetGum** [comment]
Just stumbled upon this... useful for anyone wanting to setup multiple languages with Yellow: https://github.com/datenstrom/yellow-extensions/tree/master/source/core#settings

Cheers,

Carsten

**Reply by annaesvensson** [reply]
That documentation is now at https://datenstrom.se/yellow/help/how-to-customise-languages
